vm/vm.go

```diff
@@ -257,20 +263,38 @@ func (vm *VM) Initialize(
 		return err
 	}
 	if has { //nolint:nestif
-		blkID, err := vm.GetLastAccepted()
+		statefulGenesis, err := vm.GetGenesis()
+		if err != nil {
+			snowCtx.Log.Error("could not get genesis", zap.Error(err))
+			return err
+		}
+		genesisBlk, err := chain.ParseStatefulBlock(ctx, statefulGenesis, nil, choices.Accepted, vm)
+		if err != nil {
+			snowCtx.Log.Error("could not parse genesis", zap.Error(err))
+			return err
+		}
+		vm.genesisBlk = genesisBlk
+		lastAcceptedHeight, err := vm.GetLastAcceptedHeight()
 		if err != nil {
 			snowCtx.Log.Error("could not get last accepted", zap.Error(err))
 			return err
 		}
-
-		blk, err := vm.GetStatelessBlock(ctx, blkID)
+		statefulBlock, err := vm.GetDiskBlock(lastAcceptedHeight)
 		if err != nil {
-			snowCtx.Log.Error("could not load last accepted", zap.Error(err))
+			snowCtx.Log.Error("could not get last accepted block", zap.Error(err))
 			return err
 		}
-
-		vm.preferred, vm.lastAccepted = blkID, blk
-		snowCtx.Log.Info("initialized vm from last accepted", zap.Stringer("block", blkID))
+		blk, err := chain.ParseStatefulBlock(ctx, statefulBlock, nil, choices.Accepted, vm)
+		if err != nil {
+			snowCtx.Log.Error("could not parse last accepted", zap.Error(err))
+			return err
+		}
+		vm.preferred, vm.lastAccepted = blk.ID(), blk
+		if err := vm.loadAcceptedBlocks(ctx); err != nil {
```

Comment:
Because we store the `AcceptedBlockWindow` in memory, we don't need an index of BlockID -> Height/Block.